### PR TITLE
Add support for extracting styles to tss

### PR DESCRIPTION
**Pull request by ewanharris**
Implements a new [refactoring action](https://code.visualstudio.com/docs/editor/refactoring#_refactoring-actions) for views that allows extracting the properties from a tag and inserting them into the related TSS document. This can be triggered via the usual refactor shortcut, menu item, or lightbulb. 

Some known todos (hence draft status) are:

* ~Support nested properties (font only?)~
* ~Multiline properties~
* Better handling of class merging (currently only merging classes when performing whole line extraction)
* Tests!

A demonstration is below, it first demonstrates extracting a select group of properties and then extract properties from the current line the cursor is on.

https://github.com/tidev/vscode-titanium/assets/8705251/8f9bb5a3-6ab9-435c-88d2-af9b6235939a

### CODE REVIEW

**Comment by hansemannn** [comment]
This is awesome! Regarding nested properties: All that we know are font and center (x and y). @m1ga Do you maybe have more? I wonder if they are somehow registered in Alloy already somewhere.

**Comment by m1ga** [comment]
Nice feature :+1: 

@hansemannn sorry, don't know. Didn't even know that the font part was merged :smile: Not using that many inline styles at all. It looks like it is this PR https://github.com/tidev/alloy/pull/765 and I would say that it supports all `name.property=""`

**Comment by hansemannn** [comment]
Yep, handling them in general (detect a `.` inside an attribute) might be the best way to extract those. But handling those two manually for a PoC would also be more than sufficient.

**Comment by ewanharris** [comment]
@hansemannn I've ticked off everything in that list other than the spaces/tabs one, that'll need some digging into the vscode docs to see if I can get that information or if there's a way to have them handle it when applying the edits.

File: src/providers/index.ts [hunk 1]
```diff
@@ -211,6 +223,14 @@ export function registerProviders(context: vscode.ExtensionContext): void {
 	});
 }
 
+function wrapValue(value: string|number, quote: string) {
+	if (typeof value !== 'string' || (value.startsWith('Alloy.') || value.startsWith('Ti.') || value.startsWith('Titanium.'))) {
+		return value;
```
Comment (hansemannn) [inline src/providers/index.ts:228]: $.args as well 😊

**Review by hansemannn — APPROVED** [review]
The code looks good, the changes work very well! Just to be sure: Is the version bumped automatically, or should it be adjusted in the package.json?

**Pull request merged**